File: Assignment37_1.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : ChkChar
// Description   : Check whether given character is present in string or not
// Input         : String, Character
// Output        : TRUE / FALSE
// Author        : Raj Samir Jadhav
// Date          : 01/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    public boolean ChkChar(String str, char ch)
    {
        for(int i = 0; i < str.length(); i++)
        {
            if(str.charAt(i) == ch)      // Match found
            {
                return true;
            }
        }
        return false;                    // Not found
    }
}

class Assignment37_1
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string : ");
        String input = sobj.nextLine();

        System.out.println("Enter the character : ");
        char cValue = sobj.next().charAt(0);

        Pattern pobj = new Pattern();
        boolean bRet = pobj.ChkChar(input, cValue);

        if(bRet == true)
        {
            System.out.println("TRUE");
        }
        else
        {
            System.out.println("FALSE");
        }

        sobj.close();
    }
}

File: Assignment37_1.c
#include <stdio.h>

#define TRUE 1
#define FALSE 0

typedef int BOOL;

///////////////////////////////////////////////////////////////////////////////
// Function Name : ChkChar
// Description   : Check whether the given character is present in string
// Input         : String, Character
// Output        : TRUE / FALSE
///////////////////////////////////////////////////////////////////////////////
BOOL ChkChar(char *str, char ch)
{
    while(*str != '\0')                 // Traverse string
    {
        if(*str == ch)                  // Compare each character
        {
            return TRUE;                // Found
        }
        str++;
    }
    return FALSE;                        // Not found
}

int main()
{
    char arr[50];
    char cValue;
    BOOL bRet = FALSE;

    printf("Enter string : ");
    scanf("%[^'\n']s",arr);

    printf("Enter the character : ");
    scanf(" %c",&cValue);                 // Space before %c to avoid newline issue

    bRet = ChkChar(arr,cValue);

    if(bRet == TRUE)
    {
        printf("TRUE\n");
    }
    else
    {
        printf("FALSE\n");
    }

    return 0;
}
